cl/tests: deflake replication tests

bharathv opened the pull request:
Recent changes caused some tests to flake, see individual commits.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [x] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes
* none

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

bharathv committed:
cl/tests: remove status polling thread from failures test

The poller is expected to run into errors due to failure injection.

bharathv committed:
cl/tests: increase fetch status timeouts

Due to failure injection, a subset of partitions may be leaderless which
results in them not being in the status report. Bump the timeout to
factor the delays from failures.

bharathv committed:
cl/service: tighten preconditions for report generation

Couple of additional checks.

- Ensure the service itself is initialized.
- Ensure the local manager points to a valid instance.

bharathv commented:
/ci-repeat 1
skip-units
dt-repeat=30
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkingReplicationTests

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--46d4f1e9-2a37-4438-a63b-cef3556549a8-->
<!--TEST RESULTS-->
<!--8c7683ff-e5ab-4162-a486-971e51a42cf7-->
<!--TEST RESULTS-->
<!--7837efbf-9609-4fae-9ff4-a2093ed0f996-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74494">build#74494</a></summary>


| test_class                    | test_method                    | test_arguments                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                               |
|:------------------------------|:-------------------------------|:----------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingReplicationTests | test_replication_basic         | {"shuffle_leadership": true, "source_cluster_spec": {"cluster_type": "redpanda"}} | integration | https://buildkite.com/redpanda/redpanda/builds/74494#0199fad9-0d36-4945-a4a4-8f9874d347ae | FLAKY         | 40/50    | upstream reliability is '94.89603024574669'. current run reliability is '60.0'. drift is 34.89603 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_basic         |
| ShadowLinkingReplicationTests | test_replication_basic         | {"shuffle_leadership": true, "source_cluster_spec": {"cluster_type": "redpanda"}} | integration | https://buildkite.com/redpanda/redpanda/builds/74494#0199fad9-9fc7-45b9-8a03-5a139ca206ce | FLAKY         | 46/50    | upstream reliability is '94.89603024574669'. current run reliability is '82.6086956521739'. drift is 12.28733 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_basic         |
| ShadowLinkingReplicationTests | test_replication_with_failures | null                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/74494#0199fad9-0d36-4945-a4a4-8f9874d347ae | FLAKY         | 47/50    | upstream reliability is '100.0'. current run reliability is '86.95652173913044'. drift is 13.04348 and the allowed drift is set to 50. The test should PASS            | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_with_failures |
| ShadowLinkingReplicationTests | test_replication_with_failures | null                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/74494#0199fad9-9fc7-45b9-8a03-5a139ca206ce | FLAKY         | 44/50    | upstream reliability is '97.77777777777777'. current run reliability is '75.0'. drift is 22.77778 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_with_failures |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74529">build#74529</a></summary>


| test_class               | test_method                                      | test_arguments                  | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                                            |
|:-------------------------|:-------------------------------------------------|:--------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AuditLogTestKafkaApi     | test_no_auth_enabled                             | {"audit_transport_mode": "rpc"} | integration | https://buildkite.com/redpanda/redpanda/builds/74529#019a026c-0d66-46ca-9757-056638d3b350 | FLAKY         | 18/21    | upstream reliability is '100.0'. current run reliability is '85.71428571428571'. drift is 14.28571 and the allowed drift is set to 50. The test should PASS            | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=AuditLogTestKafkaApi&test_method=test_no_auth_enabled                                 |
| NodesDecommissioningTest | test_recommissioning_one_of_decommissioned_nodes | {"cloud_topic": false}          | integration | https://buildkite.com/redpanda/redpanda/builds/74529#019a0265-6c51-44b5-98f5-36442b154426 | FLAKY         | 20/21    | upstream reliability is '99.38271604938271'. current run reliability is '95.23809523809523'. drift is 4.14462 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodesDecommissioningTest&test_method=test_recommissioning_one_of_decommissioned_nodes |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74565">build#74565</a></summary>


| test_class                          | test_method                                             | test_arguments                              | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                                   |
|:------------------------------------|:--------------------------------------------------------|:--------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| NodesDecommissioningTest            | test_decommissioning_finishes_after_manual_cancellation | {"cloud_topic": true, "delete_topic": true} | integration | https://buildkite.com/redpanda/redpanda/builds/74565#019a03c0-d8ba-431d-a467-2c2d09eb061b | FLAKY         | 20/21    | upstream reliability is '98.63013698630137'. current run reliability is '95.23809523809523'. drift is 3.39204 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodesDecommissioningTest&test_method=test_decommissioning_finishes_after_manual_cancellation |
| WriteCachingFailureInjectionE2ETest | test_crash_all                                          | {"use_transactions": false}                 | integration | https://buildkite.com/redpanda/redpanda/builds/74565#019a03c7-42e1-490e-be3c-44592a1562c7 | FLAKY         | 19/21    | upstream reliability is '82.34510326449033'. current run reliability is '90.47619047619048'. drift is -8.13109 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionE2ETest&test_method=test_crash_all                               |
</details>
<!--/body-->

bharathv commented:
~I'll follow up with a separate fix for remaining flakes (as flagged by the repeater above). The issue there is a separate bug and not a flaky test.~

update: included the fix in the PR.

bharathv committed:
k/fetch: fix fetch_session c'tor

Initialize the start_offset as seen by the request and not to default.
There are no correctness implications without this but its good to
correctly initialize for the protocol to send updates as documented.

bharathv committed:
k/fetch: fix incorrect partition inclusion on fetch retries

When fetches are retried due to byte quotas or other limits not being
reached, the inclusion decision for partitions may be incorrectly
overridden.

In this case, a partition was initially included because its
start_offset had changed. However, during a subsequent retry, the
partition was excluded because:

1. The start_offset appeared unchanged—now compared against updated
   metadata after the previous inclusion.
2. There was no data available.

As a result, the dc/fetcher no longer receives partition start offsets,
and the partition metadata is missing in the link status reports.

bharathv commented:
/ci-repeat 1
skip-units
dt-repeat=30
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkingReplicationTests

bharathv commented:
/ci-repeat 1
